File: TestFiles/sample.swift
import Foundation

// Sample Swift code for testing syntax highlighting
class HelloWorld {
    let message: String = "Hello, Prism!"
    var count: Int = 0

    func greet() {
        print(message)
        count += 1
    }

    func greetMultiple(times: Int) {
        for i in 0..<times {
            print("\(i + 1): \(message)")
        }
    }
}

// Protocol definition
protocol Printable {
    func printDescription()
}

// Extension
extension HelloWorld: Printable {
    func printDescription() {
        print("HelloWorld with message: \(message)")
    }
}

// Enum
enum Status {
    case active
    case inactive
    case pending
}

// Test usage
let hello = HelloWorld()
hello.greet()
hello.greetMultiple(times: 3)

File: TestFiles/sample.rs
// Sample Rust code for testing syntax highlighting

use std::fmt;

// Struct definition
struct HelloWorld {
    message: String,
    count: u32,
}

impl HelloWorld {
    // Constructor
    fn new(message: &str) -> Self {
        HelloWorld {
            message: message.to_string(),
            count: 0,
        }
    }

    // Method
    fn greet(&mut self) {
        println!("{}", self.message);
        self.count += 1;
    }

    fn greet_multiple(&mut self, times: u32) {
        for i in 0..times {
            println!("{}: {}", i + 1, self.message);
        }
    }
}

// Trait definition
trait Printable {
    fn print_description(&self);
}

impl Printable for HelloWorld {
    fn print_description(&self) {
        println!("HelloWorld with message: {}", self.message);
    }
}

// Enum
enum Status {
    Active,
    Inactive,
    Pending,
}

// Function
fn add(a: i32, b: i32) -> i32 {
    a + b
}

// Main function
fn main() {
    let mut hello = HelloWorld::new("Hello, Prism!");
    hello.greet();
    hello.greet_multiple(3);

    let sum = add(5, 7);
    println!("Sum: {}", sum);

    let status = Status::Active;
    match status {
        Status::Active => println!("Active"),
        Status::Inactive => println!("Inactive"),
        Status::Pending => println!("Pending"),
    }
}
